UpdatePipelineLabel/UpdatePipelineLabel/LabelUpdateHelper.cs

```diff
@@ -0,0 +1,197 @@
+            {
+                //only test for issues which has the issue number in the range
+                if ((issue.Number < issueNumberFrom || issue.Number > issueNumberTo) && !processAll)
+                {
+                    continue;
+                }
+
+                List<string> toBeRemovedLabels = new List<string>();
+                List<string> toBeAddedLabels = new List<string>();
+
+                string rightLabel;
+
+                //if the issue could not be found from zenhub's any pipeline, write an exception message and skip this issue.
+                if (!issueLabelMap.TryGetValue(issue.Number, out rightLabel))
+                {
+                    errorCount++;
+                    Console.WriteLine($"Exception : Issue {issue.HtmlUrl} could not be found in any pipeline in ZenHub");
+                }
+                else
+                {
+                    if (issue.Labels == null)
+                    {
+                        toBeAddedLabels.Add(rightLabel);
+                    }
+                    else
+                    {
+                        if (!issue.Labels.Any(e => e.Name.Equals(rightLabel)))
+                        {
+                            toBeAddedLabels.Add(rightLabel);
+                        }
+
+                        foreach (var pipelineLabel in pipelineLabels)
+                        {
+                            //if the issue has a wrong pipeline label, add it to remove list
+                            if (!pipelineLabel.Equals(rightLabel) & issue.Labels.Any(e => e.Name.Equals(pipelineLabel)))
+                            {
+                                toBeRemovedLabels.Add(pipelineLabel);
+                            }
+                        }
+                    }
+                }
+
+                //update labels
+                if (toBeAddedLabels.Count != 0 || toBeRemovedLabels.Count != 0)
+                {
+                    var issueUpdate = issue.ToUpdate();
+
+                    foreach (var label in toBeAddedLabels)
+                    {
+                        issueUpdate.AddLabel(label);
+                    }
+
+                    foreach (var label in toBeRemovedLabels)
+                    {
+                        issueUpdate.RemoveLabel(label);
+                    }
+
+                    try
+                    {
+                        await GitHubClient.Issue.Update(repoParts[0], repoParts[1], issue.Number, issueUpdate);
+                        Console.WriteLine($"Updated : issue {issue.HtmlUrl} ");
+                        updatedCount++;
+
+                        //According to https://developer.github.com/v3/guides/best-practices-for-integrators/#dealing-with-abuse-rate-limits
+                        //If you're making a large number of POST, PATCH, PUT, or DELETE requests for a single user or client ID, wait at least one second between each request.
+                        System.Threading.Thread.Sleep(1000);
```

Comment:
I think you can safely remove this -- the authenticated API limit is 5k requests per hour. We could run this script multiple times an hour and still not hit that cap: https://developer.github.com/v3/#rate-limiting

Reply:
Thanks, but I'm not sure if this is an additional limitation for PATCH request: (updating label is a PATCH request)
https://developer.github.com/v3/guides/best-practices-for-integrators/#dealing-with-abuse-rate-limits
`If you're making a large number of POST, PATCH, PUT, or DELETE requests for a single user or client ID, wait at least one second between each request.`

Reply:
Are you running into actual errors, then?

Reply:
Thanks Kat! I tested with 100+ updates without sleep, and it succeeded. Removed sleep.